ultravox/training/train.py

```diff
@@ -166,9 +152,10 @@ def train(args: config_base.TrainConfig):
         logging.info(f"Loading model state dict from {args.model_load_dir}")
         load_path = args.model_load_dir
         if wandb_utils.is_wandb_url(load_path):
-            # Download the model from W&B. The main process should do the download while the others wait.
-            with ddp_utils.run_on_master_first(is_master):
-                load_path = wandb_utils.download_model_from_wandb(load_path)
+            # We assume that the weights are already downloaded via prefetch_weights.py
+            # and hence this is just resolving the path. If the weights are not downloaded,
+            # we might see a race condition here when using DDP.
+            load_path = wandb_utils.download_model_from_wandb(load_path)
```

Comment:
Is it worth changing this API to essentially only resolve the path then, to avoid errors?

Reply:
Yeah I was thinking of that. I don't remember seeing that option in W&B API before. I will take another look.
I was thinking that keeping it won't hurt since it still supports non-DDP usages, specially if someone forgets to do prefetch_weights.
Also, IIRC W&B generally did a good job handling the race conditions even in DDP (we had mainly added `run_on_master_first` for HF downloads).

Reply:
I can see a `self._default_root()` to resolve the path but I'd rather not rely on a private method.
I also see a `should_download_entry(entry)`, but that also requires looping over contents.

I don't see an option that I would consider better I think.

Reply:
I think I miss something here. Why is it necessary to remove the `with ddp_utils.run_on_master_first(is_master)` statement?

Reply:
It's not necessary to remove it. There's just no reason for having it and it will make the start time of our training runs slower.

Reply:
The reason we had it in the first place was to avoid race-conditions when downloading the model.

Reply:
I added a prefetch_weights in train.py too to make local/test runs easier.
I'm not seeing any response on this comment. So I'm gonna assume it's resolved.